### PR TITLE
Add vsock concatenate sockets test with nested vms

### PR DESCRIPTION
**Pull request by qcheng-redhat**
ID: 2117548

Signed-off-by: qcheng <qcheng@redhat.com>

### CODE REVIEW

**Comment by qcheng-redhat** [comment]
(1/1) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.nested_vsock_con_sockets.q35: PASS (238.91 s)

**Comment by qcheng-redhat** [comment]
@PaulYuuu Could you help review this patch when you have time. Thanks.

**Comment by qcheng-redhat** [comment]
Test result on rhel8:
 (1/1) Host_RHEL.m8.u7.product_rhel.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.nested_vsock_con_sockets.q35: PASS (218.37 s)

**Comment by XueqiangWei** [comment]
@PaulYuuu Could you help review this patch. Many thanks.

File: qemu/tests/nested_vsock_con_sockets.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    playbook_repo = params["playbook_repo"]
+    playbook_timeout = params.get_numeric("playbook_timeout")
+    playbook_dir = params.get("playbook_dir",
+                              os.path.join(test.workdir, "ansible_playbook"))
+    toplevel_playbook = os.path.join(playbook_dir, params["toplevel_playbook"])
+    # Use this directory to copy some logs back from the guest
+    test_harness_log_dir = test.logdir
+
+    mq_port = params.get_numeric("mq_listen_port", 2000)
+    guest_ip_list = [vm.get_address()]
```
[the remark this answers is not in the document]
Comment (qcheng-redhat) [reply qemu/tests/nested_vsock_con_sockets.py]: Updated.

File: qemu/tests/nested_vsock_con_sockets.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
+import json
+import os
+import time
+
+from avocado.utils import path
+from avocado.utils import process
+
+from provider import message_queuing, ansible
```
[the remark this answers is not in the document]
Comment (qcheng-redhat) [reply qemu/tests/nested_vsock_con_sockets.py]: Updated.

File: qemu/tests/cfg/nested_vsock_con_sockets.cfg [hunk 3]
```diff
@@ -0,0 +1,21 @@
+- nested_vsock_con_sockets:
+    only Linux
+    type = nested_vsock_con_sockets
+    toplevel_playbook = "test.yml"
```
[the remark this answers is not in the document]
Comment (qcheng-redhat) [reply qemu/tests/cfg/nested_vsock_con_sockets.cfg]: updated.

**Review by PaulYuuu — APPROVED** [review]

**Comment by PaulYuuu** [comment]
Please provide the latest result, thanks.

**Comment by qcheng-redhat** [comment]
> Please provide the latest result, thanks.

RHEL 8:
(1/1) Host_RHEL.m8.u7.product_rhel.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.7.0.x86_64.io-github-autotest-qemu.nested_vsock_con_sockets.q35: PASS (225.30 s)

RHEL 9:
(1/1) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.1.0.x86_64.io-github-autotest-qemu.nested_vsock_con_sockets.q35: PASS (216.10 s)

**Pull request merged**